application/src/main/java/org/thingsboard/server/service/entitiy/EntityStateSourcingListener.java

```diff
@@ -247,6 +261,14 @@ private void onEdgeEvent(TenantId tenantId, EntityId entityId, Object entity, Co
         }
     }
 
+    private void onEdgeDelete(TenantId tenantId, EdgeId edgeId) {
+        if (kafkaSettings.isPresent() && kafkaTopicConfigs.isPresent()) {
+            String topic = topicService.buildEdgeEventNotificationsTopicPartitionInfo(tenantId, edgeId).getTopic();
+            TbKafkaAdmin kafkaAdmin = new TbKafkaAdmin(kafkaSettings.get(), kafkaTopicConfigs.get().getEdgeEventConfigs());
+            kafkaAdmin.deleteTopic(topic);
```

Comment:
Consumer may still be running at this point. Delete the topic only when the consumer is stopped.